src/db.c

```diff
@@ -2444,6 +2444,14 @@ int keyIsExpired(redisDb *db, sds key, kvobj *kv) {
     return now > when;
 }
 
+/* Check if user configuration allows key to be deleted due to expiary */
+int confAllowsExpireDel() {
+    if (server.allow_arbitrary_lazyexpires)
+        return 1;
+
+    return !(server.execution_nesting > 1 && server.executing_client->cmd->flags & CMD_TOUCHES_ARBITRARY_KEYS);
```

Comment:
let's add a comment here explaining the nesting thing, and also providing details about the problem.
i.e. this problem and config isn't documented anywhere, maybe here is the right place to do that (since i don't want a comment that's too long in the header).